test/Microsoft.Extensions.Configuration.Test/ConfigurationTest.cs

```diff
@@ -92,14 +92,65 @@ public void AsEnumerateFlattensIntoDictionaryTest()
             var config = configurationBuilder.Build();
 
             var dict = config.AsEnumerable().ToDictionary(k => k.Key, v => v.Value);
-            Assert.Equal("ValueInMem1", config["Mem1:KeyInMem1"]);
-            Assert.Equal("ValueDeep1", config["Mem1:KeyInMem1:Deep1"]);
-            Assert.Equal("ValueInMem2", config["Mem2:KeyInMem2"]);
-            Assert.Equal("ValueDeep2", config["Mem2:KeyInMem2:Deep2"]);
-            Assert.Equal("ValueInMem3", config["Mem3:KeyInMem3"]);
-            Assert.Equal("ValueDeep3", config["Mem3:KeyInMem3:Deep3"]);
+            Assert.Equal("ValueInMem1", dict["Mem1:KeyInMem1"]);
+            Assert.Equal("ValueDeep1", dict["Mem1:KeyInMem1:Deep1"]);
+            Assert.Equal("ValueInMem2", dict["Mem2:KeyInMem2"]);
+            Assert.Equal("ValueDeep2", dict["Mem2:KeyInMem2:Deep2"]);
+            Assert.Equal("ValueInMem3", dict["Mem3:KeyInMem3"]);
+            Assert.Equal("ValueDeep3", dict["Mem3:KeyInMem3:Deep3"]);
         }
 
+        [Fact]
+        public void AsEnumerateStripsKeyFromChildren()
```

Comment:
Are you guys talking about a value at the section where the call is made? If yes, it is interesting indeed. I don't think we ever look for this at the actual configuration root, and I am not sure `AsEnumerable()` should even return it when called on a section. Does it do it now? Re empty string:value, be careful, that empty string is a valid child key. The path of the root is null, not empty string.